docker/manage

```diff
@@ -346,27 +352,50 @@ function setNgrokEndpoints() {
   fi
 
   echoInfo "Determining ngrok url for controller service..."
-  docker run --rm curlimages/curl -L -s http://${DOCKERHOST}:4056/api/tunnels > controller-ngrok.json
-  NGROK_CONTROLLER_URL=$(docker run --rm -i stedolan/jq < controller-ngrok.json --raw-output '.tunnels | map(select(.name=="command_line")) | .[0] | .public_url')
-  if [ -z "$NGROK_CONTROLLER_URL" ]; then
-    echoError "It was not possible to establish a connection with ngrok, please check the ngrok container logs for errors."
-    exit 1
-  fi
+  getNgrokUrl http://${DOCKERHOST}:4056/api/tunnels controller-ngrok.json NGROK_CONTROLLER_URL
   export CONTROLLER_URL=${NGROK_CONTROLLER_URL}
   echoSuccess "The controller url is: ${NGROK_CONTROLLER_URL}"
 
   echoInfo "Determining ngrok url for agent service..."
-  docker run --rm curlimages/curl -L -s http://${DOCKERHOST}:${NGROK_AGENT_PORT}/api/tunnels > agent-ngrok.json
-  NGROK_AGENT_URL=$(docker run --rm -i stedolan/jq < agent-ngrok.json --raw-output '.tunnels | map(select(.name=="command_line")) | .[0] | .public_url')
-  if [ -z "$NGROK_AGENT_URL" ]; then
-    echoError "It was not possible to establish a connection with ngrok, please check the ngrok container logs for errors."
-    exit 1
-  fi
+  getNgrokUrl http://${DOCKERHOST}:${NGROK_AGENT_PORT}/api/tunnels agent-ngrok.json NGROK_AGENT_URL
   export AGENT_ENDPOINT=${NGROK_AGENT_URL}
   echoSuccess "The agent url is: ${NGROK_AGENT_URL}"
+}
+
+function getNgrokUrl() {
+  _url=$1
+  _output_file=$2
+  _target_variable=$3
 
-  rm *-ngrok.json
+  function extractUrl() {
+    docker run --rm curlimages/curl -L -s $_url > $_output_file
+    NGROK_URL=$(docker run --rm -i ghcr.io/jqlang/jq:1.7rc1 < $_output_file --raw-output '.tunnels | map(select(.name=="command_line")) | .[0] | .public_url')
+  
+    if [ -z "${NGROK_URL}" ] || [ "null" = "${NGROK_URL}" ]; then
+      return 1
+    else
+      return 0
+    fi
+  }
+
+  local startTime=${SECONDS}
+  while ! extractUrl; do
+    printf "."
+    local duration=$(($SECONDS - $startTime))
+    if (( ${duration} >= 60 )); then
```

Comment:
I would make the timeout configurable with a 60 second default.

Reply:
I thought about that, it seemed though like 60s would be the worst case scenario and in most cases it should really work after a 5-6 seconds so adding another configuration setting might be overkill?